internal/command/status/machines.go

```diff
@@ -112,20 +112,17 @@ func renderMachineStatus(ctx context.Context, app *api.AppCompact) error {
 	managed, unmanaged := []*api.Machine{}, []*api.Machine{}
 
 	for _, machine := range machines {
+		version := getFromMetadata(machine, api.MachineConfigMetadataKeyFlyPlatformVersion)
 
-		if machine.Config != nil && machine.Config.Metadata != nil {
-
-			if machine.Config.Metadata[api.MachineConfigMetadataKeyFlyPlatformVersion] == api.MachineFlyPlatformVersion2 {
-				managed = append(managed, machine)
-			} else {
-				unmanaged = append(unmanaged, machine)
-			}
-
+		if version == api.MachineFlyPlatformVersion2 {
```

Comment:
How about adding a `machine.IsAppsV2()` helper method on `api.Machine`? We'll probably end up with it in several places around the codebase.

If you update from the master branch, we now have some similar helper methods on api.Machine like IsFlyAppsPlatform().